Esri/arcgis-maps-sdk-swift-samples
# [Update] Add "No Result" text to search

## Description

This PR updates the search in the Sample Viewer to show "No Results" when applicable. This helps the search better emulate some of Apple's apps. `scrollDismissesKeyboard(.immediately)` was added for the same reason.

**Note:** The search was moved into the "All" category in #388.

## Linked Issue(s)

- N/A (quick update)

## How To Test

- Enter a query with no matching results to see the text.
- Ensure the search otherwise works as expected.

## Screenshots

|Before|After|
|:-:|:-:|
|    ![Simulator Screenshot - iPhone 15 Pro - 2024-04-18 at 13 50 38](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/70fdef62-1b56-4158-aa26-c5a4e841040e)    |    ![Simulator Screenshot - iPhone 15 Pro - 2024-04-18 at 13 51 04](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/5fd8cb42-0c0c-4188-bb4c-f68a01dea4e7)    |
